src/thunks/selectedRevisionsThunk.ts

```diff
@@ -0,0 +1,31 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { treeherderBaseURL } from '../common/constants';
+import type { APIPushResponse } from '../types/api';
+import type { Revision } from '../types/state';
+
+export const fetchSelectedRevisions = createAsyncThunk<
+  Revision,
+  { repo: string; rev: string },
+  { rejectValue: string }
+>(
+  'selectedRevisions/fetchSelectedRevisions',
+  async ({ repo, rev }, { rejectWithValue }) => {
+    let response;
+
+    try {
+      response = await fetch(
+        `${treeherderBaseURL}/api/project/${repo}/push/?revision=${rev}`,
```

Comment:
When I see a concatenation like this, I think we need to check that `repo` and `rev` can be only alphanumeric strings + comma like they should, because my understanding is that they can be changed easily in the URL.